### PR TITLE
internal-docs: list buildkit-fork features

### PR DESCRIPTION
**Pull request by alexcb**
A rough list of our features in our buildkit fork, which are not appropriate for upstream submissions.

### CODE REVIEW

**Comment by alexcb** [comment]
Here's just a starting point for documenting why we have a fork of buildkit, rather than just using upstream buildkit.

This is a very quick rough list, but figured it's good to get this initially documented, and we could expand this to individual sections later (if we feel the need to justify why we have these features, and why they cant be submitted upstream).

File: docs-internals/buildkit-fork.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Why do we have a buildkit fork?
+
+Here's a very rough list on some of the features we have in our BuildKit fork, which are too-specific to earthly,
+and would not be a good fit to submit upstream.
+
+- ability to pass arbitrary sockets from the host to buildkitd container; used by the interactive debugger
+- to support `WITH DOCKER` via host-bind mounts
+- faster exports of images from remote buildkit instance to local host; used for "pull ping" registry
```
Comment (vladaionescu) [inline docs-internals/buildkit-fork.md:8]:
```suggestion
- faster exports of images and artifacts from remote buildkit instance to local host. This includes the Earthly exporter, the pull-ping call, the embedded registry, and the storage driver to plug the registry into the buildkit cache
```

File: docs-internals/buildkit-fork.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+- git LFS support
+- verbose debugging of which files are sent to buildkit (via fsutils fork)
+- `Export` method on gateway client, which is used to enable WAIT / END blocks
+- healthcheck overrides, which are required for maintaining stability of our satellite instances
+- modifications to the `llbsolver/ops/exec.go` in order to support `LOCALLY` mode
```
Comment (vladaionescu) [inline docs-internals/buildkit-fork.md:13]:
```suggestion
- modifications to the `llbsolver/ops/exec.go` in order to support `LOCALLY` mode
- GCAnalytics for stats on garbage collection and disk space used
- info call that includes the op load and the number of sessions currently executing
- StopIfIdle - the ability to shut down buildkit while ensuring that no session is active and that no session is accepted while closing
- session management capabilities, including the ability to sessions that run for too long
```

**Review by vladaionescu — COMMENTED** [review]
Added a few more items...

**Comment by alexcb** [comment]
> Added a few more items...

thanks!

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**